x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/total_feature_importance_summary/feature_importance_summary.tsx

```diff
@@ -189,13 +194,52 @@ export const FeatureImportanceSummaryPanel: FC<FeatureImportanceSummaryPanelProp
   const tickFormatter = useCallback((d) => Number(d.toPrecision(3)).toString(), []);
 
   // do not expand by default if no feature importance data
-  const hasTotalFeatureImportance = useMemo(() => totalFeatureImportance.length > 1, [
-    totalFeatureImportance,
-  ]);
+  const noDataCallOut = useMemo(() => {
+    // if no total feature importance data
+    if (totalFeatureImportance.length === 0) {
+      // check if it's because num_top_feature_importance_values is set to 0
+      if (
+        (jobConfig?.analysis && isRegressionAnalysis(jobConfig?.analysis)) ||
+        isClassificationAnalysis(jobConfig?.analysis)
+      ) {
+        const analysisType = getAnalysisType(jobConfig.analysis);
```

Comment:
Nit: Might be worth moving `const analysisType = getAnalysisType(jobConfig.analysis);` above the if block since you need to access to the analysis type to check if it's regression or classification. Then you could use `ANALYSIS_CONFIG_TYPE` to check if `analysisType === ANALYSIS_CONFIG_TYPE.CLASSIFICATION` etc.

Reply:
Unfortunately this is a typescript issue. I had to add the extra isRegressionAnalysis and isClassificationAnalysis check so we can access `jobConfig.analysis[analysisType].num_top_feature_importance_values`, so I'll keep it for now.